endpoints/users_groups/src/main.cpp

```diff
@@ -107,6 +109,70 @@ namespace irods::http::handler
 
 namespace
 {
+	//
+	// Backward compatibility implementations
+	//
```

Comment:
Would it make sense to put this in the `irods::http::compatibility` namespace? I realize we are trying to not allow linking this externally, but we could make it `static`. This would become self-explanatory (IMO) and we could get rid of the `compat_` prefix on the function name below. Not a major thing.

Reply:
I've been going back and forth about that in my head, but haven't reached a decision.

I may end up moving it into the compatibility source files. I like the idea of people being able to see all of the compatibility logic in one place. On the flip side, I also like keeping the code which uses the compat_add_group function in the same source file because it's only used in one place.

Will keep weighing the the approaches. Open to hearing more thoughts about this.

Reply:
Moved the function in the compatibility namespace.

Confirmed the code changes still pass tests.